testapp/tests.py

```diff
@@ -60,86 +62,187 @@ def test_create_change_request(self, mock_get_snow_group_guid, mock_pysnow):
+            'assignment_group': {'value': uuid.uuid4()},
+            'state': '2'
+        }
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+        self.mock_pysnow_client.resource.return_value = fake_resource
+        mock_pysnow.Client.return_value = self.mock_pysnow_client
+
+        payload = OrderedDict()
+        payload['type'] = 'normal'
+        payload['assignment_group'] = 'bar'
+        self.change_request_handler.create_change_request('Title', 'Description', None, payload=payload)
+        fake_resource.create.assert_called_with(payload=expected_payload)
+
+    def test_create_change_request_default_parameters(self, mock_pysnow):
+        expected_payload = OrderedDict()
+        expected_payload['short_description'] = 'Title'
+        expected_payload['description'] = 'Description'
+        expected_payload['type'] = 'standard'
+        expected_payload['assignment_group'] = 'bar'
+
+        self.change_request_handler.group_guid_dict['assignment_group'] = 'bar'
+
+        fake_insert_retval = {
+            'sys_id': uuid.uuid4(),
+            'number': 'CHG0000001',
+            'short_description': 'Title',
+            'description': 'Description',
+            'assignment_group': {'value': uuid.uuid4()},
+            'state': '2'
+        }
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+        self.mock_pysnow_client.resource.return_value = fake_resource
+        mock_pysnow.Client.return_value = self.mock_pysnow_client
+
+        self.change_request_handler.create_change_request('Title', 'Description', None, payload=OrderedDict())
+        fake_resource.create.assert_called_with(payload=expected_payload)
+
+    def test_create_change_request_raises_exception_for_http_error(self, mock_pysnow):
+        fake_resource = mock.MagicMock()
+
+        fake_exception = HTTPError()
+        fake_exception.response = mock.MagicMock()
+        fake_exception.response.text.return_value = 'Foobar'
+
+        fake_resource.create.side_effect = fake_exception
+
+        self.mock_pysnow_client.resource.return_value = fake_resource
+        mock_pysnow.Client.return_value = self.mock_pysnow_client
+
+        with self.assertRaises(ChangeRequestException):
+            self.change_request_handler.create_change_request('Title', 'Description', None, payload={})
+
+    def test_create_change_request_raises_exception_when_error_in_result(self, mock_pysnow):
         fake_insert_retval = {
             'error': 'some error message'
         }
-        self.mock_pysnow_client.insert.return_value = fake_insert_retval
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+
+        self.mock_pysnow_client.resource.return_value = fake_resource
         mock_pysnow.Client.return_value = self.mock_pysnow_client
 
         with self.assertRaises(ChangeRequestException):
```

Comment:
Since this is checking for the exact same exception as above, generated by the same method call, I think this should also verify that the proper error message is being raised, with `six.assertRaisesRegex()`.